source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -541,6 +549,12 @@ void HappyEyeballsConnectionImpl::onBelowWriteBufferLowWatermark(
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+void HappyEyeballsConnectionImpl::onWriteBufferLowWatermark() {
+  // Only be called went moving write data from the deferred write buffer
+  // to the underlying connection. In this case, the connection callbacks must
+  // not be notified since this should be transparent to the callbacks.
```

Comment:
Along with grammar fix ups, can you comment that we also make sure that when the "high watermark" data is passed from here to the connection buffer we make sure the extra "high watermark" call gets swallowed?

Reply:
Sure! Added a comment to this effect in `setUpFinalConnection()`